actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -90,18 +103,18 @@ class ActionCable.Connection
     open: ->
       ActionCable.log("WebSocket onopen event")
       @disconnected = false
-      @subscriptions.reload()
+      if not @isProtocolSupported()
+        ActionCable.log("WebSocket got '#{@getProtocol()}' as the chosen protocol. Stopping monitor and disconnecting.")
+        @close(allowReconnect: false)
+        return
+      ActionCable.log("WebSocket using '#{@getProtocol()}' subprotocol.")
 
-    close: ->
+    close: (event) ->
       ActionCable.log("WebSocket onclose event")
-      @disconnect()
+      return if @disconnected
+      @disconnected = true
```

Comment:
Tracking `@disconnected` was originally in place because to prevent calling the `disconnect` method twice since both the `error` and `close` handlers called it. Now that `disconnect` has been removed, I think we can rely on the connection state alone and remove `@disconnected`.